ouroboros-consensus-cardano/ouroboros-consensus-cardano.cabal

```diff
@@ -662,9 +664,35 @@ test-suite tools-test
   hs-source-dirs: test/tools-test
   main-is: Main.hs
   build-depends:
+    QuickCheck,
+    aeson,
     base,
     ouroboros-consensus:{ouroboros-consensus, unstable-consensus-testlib},
-    ouroboros-consensus-cardano,
+    ouroboros-consensus-cardano:{ouroboros-consensus-cardano, unstable-cardano-tools},
+    ouroboros-consensus-protocol:unstable-protocol-testlib,
     tasty,
     tasty-hunit,
+    tasty-quickcheck,
+    text,
     unstable-cardano-tools,
+
+  other-modules:
+    Test.Cardano.Tools.Headers
+
+executable gen-header
+  import: common-exe
+  hs-source-dirs: app
+  main-is: gen-header.hs
+  build-depends:
+    base,
+    cardano-crypto-class,
+    optparse-applicative,
+    ouroboros-consensus-cardano:unstable-cardano-tools,
+    with-utf8,
+
+  other-modules:
+    GenHeader.Parsers
+    Paths_ouroboros_consensus_cardano
+
+  autogen-modules:
+    Paths_ouroboros_consensus_cardano
```

Comment:
We don't usually include the `Paths_` modules in the cabal files.

Reply:
I need this to be able to retrieve the current version to display in https://github.com/abailly/ouroboros-consensus/blob/67917c2bc6e1c3b10960a8d1933bf2e701c5ad0b/ouroboros-consensus-cardano/app/GenHeader/Parsers.hs#L19
Is there another, better, way to do this?

Reply:
Ah I didn't see that use, sorry. You are indeed right, if you need that then you need to declare the `Paths_` module. However I'm unsure if that would be the right version to consider or should you print some other version number. The library version number is not really representative of what protocol it understands.

Reply:
What would be the best option?

Reply:
Probably @amesgen is the one that knows the answer more precisely here.

Reply:
I think the version number is fine at least for now, especially as this tool is quite new and will probably be evolving for some time. We could also include the git rev:
https://github.com/IntersectMBO/ouroboros-consensus/blob/e86b921443bd6e8ea25e7190eb7cb5788e28f4cc/ouroboros-consensus-cardano/src/unstable-cardano-tools/Cardano/Tools/GitRev.hs#L11-L13

Reply:
I will do it in another PR if you don't mind, this is a first stab at generators for validation and I would rather like to have this as a baseline before adding more stuff, if possible